packages/blocks/src/generators/arduino.ts

```diff
@@ -108,11 +101,10 @@ export class Arduino extends Blockly.Generator {
 		return this.getRawPinMapping(pin);
 	}
 
-	public clearBuilder() {
-		this.builder = new ComponentBuilder();
-		this.murphy = this.builder.add("murphy", Murphy);
-		this.i2c = this.builder.add("murphy-i2c", MurphyI2C);
-		this.builder.join(this.murphy, this.i2c);
+	public createSchemaBuilder() {
+		if (this.boardType.includes("nano")) {
+			this.builder = new ComponentBuilder();
```

Comment:
this.builder is never deleted if the robot type is not nano, so switching from a nano to a mega environment will trigger the bug again